core/files/fileoffloader/offloader.go

```diff
@@ -108,30 +110,41 @@ func (s *service) FilesOffload(ctx context.Context, objectIds []string, includeN
 }
 
 func (s *service) offloadAllFiles(ctx context.Context, includeNotPinned bool) (filesOffloaded int, totalSize uint64, err error) {
-	records, err := s.objectStore.Query(database.Query{
-		Filters: []*model.BlockContentDataviewFilter{
-			{
-				RelationKey: bundle.RelationKeyFileId.String(),
-				Condition:   model.BlockContentDataviewFilter_NotEmpty,
+	gc := s.fileStorage.NewLocalStoreGarbageCollector()
+
+	if !includeNotPinned {
+		records, err := s.objectStore.Query(database.Query{
+			Filters: []*model.BlockContentDataviewFilter{
+				{
+					RelationKey: bundle.RelationKeyFileId.String(),
+					Condition:   model.BlockContentDataviewFilter_NotEmpty,
+				},
+				{
+					RelationKey: bundle.RelationKeyFileBackupStatus.String(),
+					Condition:   model.BlockContentDataviewFilter_NotEqual,
+					Value:       pbtypes.Int64(int64(filesyncstatus.Synced)),
+				},
 			},
-		},
-	})
-	if err != nil {
-		return 0, 0, fmt.Errorf("query file objects by spaceId: %w", err)
-	}
-	for _, record := range records {
-		size, err := s.fileOffload(ctx, record.Details, includeNotPinned)
+		})
 		if err != nil {
-			objectId := pbtypes.GetString(record.Details, bundle.RelationKeyId.String())
-			log.Error("offloadAllFiles: failed to offload file", zap.String("objectId", objectId), zap.Error(err))
-			continue
+			return 0, 0, fmt.Errorf("query not pinned files: %w", err)
 		}
-		totalSize += size
-		if size > 0 {
-			filesOffloaded++
+
+		for _, record := range records {
+			fileId := domain.FullFileId{
+				SpaceId: pbtypes.GetString(record.Details, bundle.RelationKeySpaceId.String()),
+				FileId:  domain.FileId(pbtypes.GetString(record.Details, bundle.RelationKeyFileId.String())),
+			}
+			_, cids, err := s.getAllExistingFileBlocksCids(ctx, fileId)
+			if err != nil {
+				return 0, 0, fmt.Errorf("not pinned file: collect cids: %w", err)
+			}
+			gc.MarkAsUsing(cids)
 		}
 	}
-	return filesOffloaded, totalSize, nil
+
+	err = gc.CollectGarbage(ctx)
+	return 0, 0, err
```

Comment:
Is it okay that we always return 0 and 0?

Reply:
Thank you. Afaik, clients don't use these fields. I'd better to remove them